dev/benchmarks/macrobenchmarks/lib/src/stack_size.dart

```diff
@@ -0,0 +1,113 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ffi' as ffi;
+import 'dart:io' as io;
+
+import 'package:flutter/material.dart';
+
+import '../common.dart';
+
+typedef GetStackPointerCallback = int Function();
+
+// c interop function:
+// void* mmap(void* addr, size_t len, int prot, int flags, int fd, off_t offset);
+typedef CMmap = ffi.Pointer<ffi.Void> Function(
+    ffi.Pointer<ffi.Void>, ffi.IntPtr, ffi.Int32, ffi.Int32, ffi.Int32, ffi.IntPtr);
+typedef DartMmap = ffi.Pointer<ffi.Void> Function(
+    ffi.Pointer<ffi.Void>, int, int, int, int, int);
+final DartMmap mmap = ffi.DynamicLibrary.process().lookupFunction<CMmap, DartMmap>('mmap');
+
+// c interop function:
+// int mprotect(void* addr, size_t len, int prot);
+typedef CMprotect = ffi.Int32 Function(ffi.Pointer<ffi.Void>, ffi.IntPtr, ffi.Int32);
+typedef DartMprotect = int Function(ffi.Pointer<ffi.Void>, int, int);
+final DartMprotect mprotect = ffi.DynamicLibrary.process()
+    .lookupFunction<CMprotect, DartMprotect>('mprotect');
+
+const int protRead = 1;
+const int protWrite = 2;
+const int protExec = 4;
+
+const int mapPrivate = 0x02;
+const int mapJit = 0x0;
+const int mapAnon = 0x20;
+
+String get currentArch {
+  return ffi.sizeOf<ffi.IntPtr>() == 4 ? 'arm' : 'arm64';
```

Comment:
Could this run on an intel-based emulator?

Reply:
I think we should just keep one cpu architecture, otherwise this benchmark does not really make sense. I will remove one of them